test/run.sh

```diff
@@ -162,6 +167,13 @@ do
          exit \$retval" >> $logfile 2>&1
   RETVAL=$?
 
+  # if the test is a benchmark we have to collect the results before removing the folder
+  if [ x"$cvmfs_benchmark" = x"yes" ]; then
+    collect_benchmark_results
+    umount /cvmfs/$FQRN > /dev/null 2>&1
```

Comment:
Can you use the umount function from the test utilities here?  The one that waits until the umount is completed?